src/lib.rs

```diff
@@ -956,8 +1032,9 @@ impl GooseAttack {
         // 'u' will always be the greatest common divisor
         debug!("gcd: {}", u);
 
-        // Build a weighted lists of task sets (identified by index)
-        let mut weighted_task_sets = Vec::new();
+        // Build a vector of vectors to be used to schedule users.
+        let mut available_task_sets = Vec::new();
```

Comment:
micro-optimization (probably not worth it): 

You could already know the total_tasks_set at this point so reserve capacity in the Vec.

Reply:
You're absolutely correct, and [the documentation makes it clear](https://doc.rust-lang.org/std/vec/struct.Vec.html#capacity-and-reallocation) it is recommended to specify capacity if you know it, otherwise the vector may get moved around in memory as it grows. This would be more important with larger load tests. The `available_task_sets` vector will always be the same length as our existing vector of `task_sets`, fixed.